src/extent.c

```diff
@@ -1930,6 +1980,11 @@ extent_merge_impl(tsdn_t *tsdn, arena_t *arena,
 		return true;
 	}
 
+	if (extent_dumpable_get(a) || extent_dumpable_get(b)) {
+		extent_ensure_dumpable(a);
+		extent_ensure_dumpable(b);
```

Comment:
Just realized, this could make no dump state go away, if we happen to release and merge with the large remaining undumpable extent from grow_retained. Is that true? If that's the case, I'm start to leaning toward having a specific extent in the arena (and never merge with it, only alloc from it when trying to grow_retain).